### PR TITLE
instrumentation/net/http/otelhttp: support duplicate both old and new attributes for HTTPClient

### PR DESCRIPTION
**Pull request by VinozzZ**
This PR implements the HTTPClient that can duplicate attributes for old semconv v1.20.0 and new semconv v1.26.0.
It's part of the migration plan #5332

### CODE REVIEW

**Review by MadVikingGod — APPROVED** [review]
Thank you for this.

The only attribute we might want to consider adding is `url.scheme`.

**Comment by VinozzZ** [comment]
> Thank you for this.
> 
> The only attribute we might want to consider adding is `url.scheme`.

Thank you for the review!
I also had a discussion at the last sig meeting about the new attribute `error.type`. 
The consensus is to implement a new method to set `error.type`. The new method will replace the `RecordError()` in `RoundTrip()` in `otelhttp/transport.go`. I will make this change soon

File: instrumentation/net/http/otelhttp/internal/semconv/httpconv.go [hunk 1]
```diff
@@ -195,3 +195,143 @@ func (n newHTTPServer) ResponseTraceAttrs(resp ResponseTelemetry) []attribute.Ke
 func (n newHTTPServer) Route(route string) attribute.KeyValue {
 	return semconvNew.HTTPRoute(route)
 }
+
+type newHTTPClient struct{}
+
+// RequestTraceAttrs returns trace attributes for an HTTP request made by a client.
+func (n newHTTPClient) RequestTraceAttrs(req *http.Request) []attribute.KeyValue {
```
Comment (dmathieu) [inline instrumentation/net/http/otelhttp/internal/semconv/httpconv.go:202]: Should those methods be tested?

Comment (VinozzZ) [reply instrumentation/net/http/otelhttp/internal/semconv/httpconv.go:205]: This is tested in the httpconv_test.go. Is there anywhere else I should add tests for this method?

File: instrumentation/net/http/otelhttp/internal/semconv/httpconv.go [hunk 2]
```diff
@@ -195,3 +198,151 @@ func (n newHTTPServer) ResponseTraceAttrs(resp ResponseTelemetry) []attribute.Ke
...
+	}
+
+	if isErrorStatusCode(resp.StatusCode) {
+		errorType := strconv.Itoa(resp.StatusCode)
+		attrs = append(attrs, semconvNew.ErrorTypeKey.String(errorType))
+	}
+	return attrs
+}
+
+func (n newHTTPClient) ErrorType(err error) attribute.KeyValue {
```
Comment (VinozzZ) [inline instrumentation/net/http/otelhttp/internal/semconv/httpconv.go:314]: I'm not sure if this is the correct implementation that comply with the spec.
In the [spec](https://opentelemetry.io/docs/specs/semconv/attributes-registry/error/), it says `Instrumentations SHOULD document the list of errors they report.`. What error types do we want to return here?

**Review by dashpole — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Pull request merged**